custreamz_notebooks/streaming_word_count.ipynb

```diff
@@ -0,0 +1,275 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Streamz and cudf imports\n",
+    "import cudf\n",
+    "from streamz import Stream\n",
+    "from streamz.dataframe import DataFrame"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's assume that the data coming in to the Kafka topic — i.e., each record/message, is a line in the form of \"this is line x\", where x is an incremental counter. \n",
+    "    \n",
+    "Now, we write a function to parse each such message to get the list of words in each line. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Helper function operating on every batch polled from Kafka for word count\n",
+    "def process_batch(messages):\n",
+    "    y = []\n",
+    "    for x in messages:\n",
+    "        y = y + x.decode('utf-8').strip('\\n').split(\" \")\n",
+    "    return y"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's create a Kafka consumer. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Kafka topic to read streaming data from\n",
+    "topic = \"word-count\"\n",
+    "\n",
+    "#Kafka brokers\n",
+    "bootstrap_servers = 'localhost:9092'\n",
+    "\n",
+    "#Kafka consumer configuration\n",
+    "consumer_conf = {'bootstrap.servers': bootstrap_servers, 'group.id': 'custreamz', 'session.timeout.ms': 60000}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now use Streamz to create a Stream from Kafka by polling the topic every 10s. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#If you changed Dask=True, please ensure you have a Dask cluster up and running\n",
+    "source = Stream.from_kafka_batched(topic, consumer_conf, npartitions=1, poll_interval='10s', asynchronous=True, dask=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Applying process_batch function to process word count on each batch\n",
+    "stream = source.map(process_batch)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Streamz DataFrame does the trick!* \n",
+    "\n",
+    "After we get the parsed word list on our stream from Kafka, we just perform simple aggregations using the Streamz DataFrame to get the word count.\n",
+    "\n",
+    "We then write the output (word count) to a list."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "stream_df = stream.map(lambda words: cudf.DataFrame({'word': words, 'count': [1 for i in words]}))\n",
```

Comment:
Not sure, if this is a bottleneck but can be made faster with below (even faster if we do a cupy array) 
```
cudf.DataFrame({'word': words, 'count':  np.ones(len(words),dtype=np.int32)})
```
`list` column assignment is not that preferment with `cudf`

Same for `words` if it is a string array. 
df['word'] = nvstrings.to_device(words)

Also, not sure why the `count` column is needed here .

Reply:
Awesome, I'll fix this in the notebook!